BrentOzarULTD/SQL-Server-First-Responder-Kit
# Updated DatabaseRestore to support versions from 2008 forward.

Altered code for capturing RESTORE statement output to use dynamic SQL and capture columns based on version number. Changed table vars to temp tables to work with the dynamic SQL.

Fixes #774 .

Changes proposed in this pull request:
 - Updated DatabaseRestore to use dynamic SQL for capturing columns from RESTORE HEADERONLY and RESTORE FILELISTONLY, allowing it to be used from SQL Server 2008 onwards.

How to test this code:
 - Obtain a backup from the version of SQL Server you want to test
 - Verify it can be restored successfully to SQL Server 2016 (regression test)
 - Verify it can be restored successfully to the target version of SQL Server (new feature test)

Has been tested on (remove any that don't apply):
 - SQL Server 2008 R2
 - SQL Server 2014
 - SQL Server 2016
